### DISCUSSION TITLE
Increase watchers in GitHub Workflow when running from a docker image

### DISCUSSION THREAD

**Discussion by hashim-sohail** [community/community#25803] [Actions]
<p>I’m currently running a  job in my workflow using my custom docker image which has all the environment setup. </p>
<p>On the step of Expo Deploy i’m running into this error <code>Error: ENOSPC: System limit for number of file watchers reached</code>. I can’t increase the watchers in my docker image as it is only available in read-only mode if not ran with privileges. </p>
<p>The job in my workflow</p>
<pre><code>expo-publish:
    name: Expo Publish
    needs: test
    runs-on: ubuntu-latest
    container:
      image: bluebasejs/github-actions-image

    strategy:
      matrix:
        node-version: [12]

    steps:
      - uses: actions/checkout@v1
      - name: Install Dependencies
        env:
          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
        run: yarn install

      - name: Build Expo
        run: HOME=/root bluebase expo:build

      - name: Expo Deploy
        uses: BlueBaseJS/expo-deploy-github-action@master
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          EXPO_CLI_USERNAME: ${{ secrets.EXPO_CLI_USERNAME }}
          EXPO_CLI_PASSWORD: ${{ secrets.EXPO_CLI_PASSWORD }}
</code></pre>
<p>Adding  this step to the job </p>
<pre><code>- run: fs.inotify.max_user_watches=524288 | tee -a /etc/sysctl.conf &amp;&amp; sysctl -p
</code></pre>
<p>results in the error </p>
<pre><code>sysctl: setting key "fs.inotify.max_user_watches": Read-only file system
</code></pre>

**Comment by ethomson** [comment, marked as answer]
<p>This is a <a>docker limitation</a>, you cannot change sysctl settings by default.</p>
<p>You may be able to change the sysctl settings using the <code>--sysctl ...</code> option, adding that to the <a>container options</a>.</p>

**Comment by hashim-sohail** [comment]
<p>Solved it by adding the  <strong>–privileged </strong> to the options</p>
<blockquote>
<p><a class="mention" href="/ethomson">@ethomson</a> wrote:</p>
</blockquote>
<blockquote>
<p>This is a <a>docker limitation</a>, you cannot change sysctl settings by default.</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>You may be able to change the sysctl settings using the <code>--sysctl ...</code> option, adding that to the <a>container options</a>.</p>
</blockquote>
